homeassistant/components/vacuum/roomba.py

```diff
@@ -186,6 +213,34 @@ def async_return_to_base(self, **kwargs):
         self._is_on = False
 
     @asyncio.coroutine
+    def async_set_fan_speed(self, fan_speed, **kwargs):
+        """Set fan speed."""
+        if fan_speed.capitalize() in FAN_SPEEDS:
+            fan_speed = fan_speed.capitalize()
+        _LOGGER.debug("Set fan speed to: %s", fan_speed)
+        high_perf = None
+        carpet_boost = None
+        if fan_speed == FAN_SPEED_AUTOMATIC:
+            high_perf = False
+            carpet_boost = True
+            self._fan_speed = FAN_SPEED_AUTOMATIC
+        elif fan_speed == FAN_SPEED_ECO:
+            high_perf = False
+            carpet_boost = False
+            self._fan_speed = FAN_SPEED_ECO
+        elif fan_speed == FAN_SPEED_PERFORMANCE:
+            high_perf = True
+            carpet_boost = False
+            self._fan_speed = FAN_SPEED_PERFORMANCE
+        else:
+            _LOGGER.error("No such fan speed available: %s", fan_speed)
+        # The set_preference method does only accept string values
```

Comment:
Don't you want to return here and not call any device methods if the fan speed argument was bad?